identity_stardust/examples/create_did.rs

```diff
@@ -114,126 +88,83 @@ async fn main() -> anyhow::Result<()> {
-
-  // Infer DID from the Alias Output block.
-  let did: StardustDID = StardustDID::from_block(&block1, &network_name)?;
-  println!("DID: {did}");
-
-  // ===========================================================================
-  // Step 3: Resolve a DID Document.
-  // ===========================================================================
-  // iota.rs indexer example:
-  // https://github.com/iotaledger/iota.rs/blob/f945ccf326829a418334942ae9cf53b8fab3dbe5/examples/indexer.rs
-
-  // Extract Alias ID from the DID tag.
-  let alias_id: AliasId = AliasId::from_str(did.tag())?;
-  println!("Alias ID: {alias_id}");
-
-  // Query Indexer INX Plugin for the Output of the Alias ID.
-  let alias_output_id = client.alias_output_id(alias_id).await?;
-  println!("Output ID: {alias_output_id}");
-  let response = client.get_output(&alias_output_id).await?;
-  let output = Output::try_from(&response.output)?;
-  println!("Output: {output:?}");
-
-  // The resolved DID Document replaces the placeholder DID with the correct one.
-  let resolved_document = StardustDocument::unpack_from_output(&did, &output)?;
-  println!("Resolved Document: {resolved_document:#}");
-
-  let alias_output = match output {
-    Output::Alias(output) => Ok(output),
-    _ => Err(anyhow::anyhow!("not an alias output")),
-  }?;
-
-  // ===========================================================================
-  // Step 4: Publish an updated Alias Output. (optional)
-  // ===========================================================================
-
-  // Add a new Ed25519 verification method to the DID Document for authentication. (optional)
-  let mut updated_document = resolved_document.clone();
-  let keypair: KeyPair = KeyPair::new(KeyType::Ed25519)?;
-  let method = StardustVerificationMethod::new(
-    updated_document.id().clone(),
-    keypair.type_(),
-    keypair.public(),
-    "#key-0",
-  )?;
-  updated_document.insert_method(method, MethodScope::authentication())?;
-
-  // Update the Alias Output to contain an explicit Alias ID, and the updated DID Document.
-  let byte_cost_config: ByteCostConfig = client.get_byte_cost_config().await?;
-  let updated_alias_output = AliasOutputBuilder::from(&alias_output)
-    // Set the deposit to the new minimum covering the increased size of the DID Document.
-    .with_minimum_storage_deposit(byte_cost_config)
-    // Set the explicit Alias ID.
-    .with_alias_id(alias_id)
-    // Set the updated DID Document.
-    .with_state_metadata(updated_document.pack()?)
-    // State controller updates must increment the state index.
-    .with_state_index(alias_output.state_index() + 1)
-    .finish_output()?;
 
-  println!("Updated output: {updated_alias_output:?}");
+  // Publish the output and get the published document.
+  let block: Block = client.publish_outputs(&secret_manager, vec![output]).await?;
+  let documents: Vec<StardustDocument> = client.documents_from_block(&block).await?;
```

Comment:
The main thought was that publishing multiple identities together would not be possible with the current API if `publish_output` would just take a single output. And in the account we might want to support that. On the flipside, extracting a single document form a vector is easily possible.
However, I agree with your other comment that building the alias output is harder than publishing so if we go down the `let alias_output = client.new_identity(&doc)` route, publishing multiple outputs in one transaction would be easily possible, since we wouldn't handle publishing.